CNV-1589: Provision and manage VMs using Ansible modules (#14914)
openshift/openshift-docs: cnv/cnv_users_guide/cnv-ansible-modules.adoc

> [id="ansible-modules"]

Reviewer: ```suggestion
// Module included in the following assemblies:
//
// *

[id="cnv-ansible-modules_{context}"]
```

Reviewer: (the _{context} part is used in modules, but it's not included in assemblies.)

> // Module included in the following assemblies:

Reviewer 2: If this file is a module, then it should go in the modules/ directory.

Reviewer: Ah, excellent catch! Piotr, please move the file to the modules directory. Thanks :)

//

> // *

Reviewer 2: And I can see that this module isn't added to an assembly yet (should it be now?), but when you're ready, then it should be updated here to list which assembly the module is a part of.

Reviewer: @bergerhoffer One of us from CNV docs is going to do a follow-up PR to create an assembly for this module.

> = Automating management tasks using Ansible

Reviewer: ```suggestion
= Automating {ProductName} management tasks by using Red Hat Ansible Automation
```

> :context: cnv-ansible-modules

Reviewer: ```suggestion
```

Reviewer: We use this in assemblies to build the direct link to the module. The [id="cnv-ansible-modules_{context}"] gets its context from the :context: line of the assembly that the module is in.

Currently, there's no assembly that includes this module, but I'll create one as a follow-up PR.

> toc::[]

Reviewer: ```suggestion
```

Reviewer: There isn't a table of contents in the module itself, so there's no need for `toc::[]`.

> = Automating {ProductName} management tasks by using Red Hat Ansible Automation

Reviewer 2: @ousleyp just a general comment (not for this PR) - this attribute name is a little confusing, since we have "product-title" for OpenShift and if I hadn't build it, I wouldn't have known that this was the CNV product name. I'd suggest to change this attribute name to CNVProductName or something, so that it's easier to understand from the source that it's CNV's name.

Reviewer: @bergerhoffer Good call, I have wondered if this is confusing. I can bring it up in our CNV meeting this week.

> include::modules/cnv-document-attributes.adoc[]

Reviewer 2: I'm not sure if how CNV does things is different, but I think once you add this module to an assembly, I'd think the assembly should include this attribtues file, and you wouldn't include it here.

> Ansible is an automation tool used to configure systems, deploy software, and perform rolling updates.

Reviewer: I'm waiting to find out if "Ansible" is okay or if the shortname should be "Ansible Automation" which is how it's written in the spreadsheet of official product names/shortnames. I'm not sure if the list is out-of-date or if it's a new convention that we should start following.

Author: In RHV [documentation](https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.3/html/administration_guide/chap-automating_rhv_configuration_using_ansible) we use `Ansible`.

Reviewer: Thanks for letting me know. I know that in the rest of the OpenShift docs, "Ansible" is used, but it's possible the "Ansible Automation" guideline is new. I'll let you know when I have confirmation.

> Ansible includes support for {ProductName}, and Ansible modules are available to allow you to automate

Reviewer: ```suggestion
Ansible includes support for {ProductName}, and Ansible modules enable you to automate
```

> cluster management tasks such as template, persistent volume claim and virtual machine operations.

Reviewer: ```suggestion
cluster management tasks such as template, persistent volume claim, and virtual machine operations.
```

> Ansible provides an alternative way pf automating {ProductName} management compared to APIs and oc tool,

Reviewer 3: pf/of

> Ansible provides an alternative way of automating {ProductName} management compared to APIs and oc tool,

Reviewer: Overall, the writing is good and you followed our conventions well (thanks for using {ProductName}! 

This particular sentence is a bit unclear to me, though. When you say it's "an alternative way" "compared to APIs and [the] `oc` tool," I'm not sure what is being compared. If you want to say that Ansible is better than using APIs or the `oc` tool, give a reason. :) Otherwise, I don't think it's necessary to mention APIs/the `oc` tool here.

Might be better to just say something like: "Ansible allows you to integrate with other Ansible modules, \<give an example or reason why you would want to do that\>."

Author: The intention was not to compare but state that it is yet another way to manage our `{ProductName}` but we have a benefit to use other modules provided by Ansible.

Ansible provides a way to automate {ProductName} management, which you can also accomplish by using

> the oc CLI tool or APIs. Ansible is unique because it allows you to integrate

Reviewer 2: oc should be in backticks here, \`oc\`

> For more information about the Ansible modules available for {ProductName}, see link:https://docs.ansible.com/ansible/latest/modules/list_of_cloud_modules.html#kubevirt[Kubevirt modules]

Reviewer: Generally we try to avoid "see this link." Maybe the previous sentence should read:

 "...allows you to integrate link:https://docs.ansible.com/ansible/latest/modules/list_of_cloud_modules.html#kubevirt[KubeVirt modules] with other Ansible modules." 

or something like that?

Author: Sure will update.

Author: @ousleyp What do you think about:

> Among other tools Ansible provides a way to automate {ProductName} management and allows to integrate link:https://docs.ansible.com/ansible/latest/modules/list_of_cloud_modules.html#kubevirt[KubeVirt modules] with other Ansible modules.

Reviewer: > Ansible provides a way to automate {ProductName} management, which you can also accomplish by using the `oc` CLI tool or APIs. Ansible is unique because it allows you to integrate link:https://docs.ansible.com/ansible/latest/modules/list_of_cloud_modules.html#kubevirt[KubeVirt modules] with other Ansible modules.

What do you think of this? If it doesn't make sense, no worries.

Author: Looks good, will update
